### PR TITLE
[3.4] Backport ignore SetKeepAlivePeriod errors on OpenBSD

### PR DESCRIPTION
**Pull request by Elbehery**
This is a backport of https://github.com/etcd-io/etcd/pull/15812

cc @ahrtr

### CODE REVIEW

**Comment by ahrtr** [comment]
PLease do not make unrelated change, you only take care of `(l *limitListenerConn) SetKeepAlivePeriod ` in the similar way https://github.com/etcd-io/etcd/blob/1a6eaca5398c7a6597786b5ae352809fc9977d1f/pkg/transport/limit_listen.go#L74-L80

**Comment by ahrtr** [comment]
Also please try not to directly cherry-pick commit from main branch to stable branches next time, because there are huge differences between main and releaase-3.5 & release-3.4.

**Comment by ahrtr** [comment]
> you only need to take care of `(l *limitListenerConn) SetKeepAlivePeriod ` in the similar way

This isn't correct. PLease ignore this comment

File: pkg/transport/keepalive_listener.go [hunk 1]
```diff
@@ -72,17 +101,12 @@ func (l *tlsKeepaliveListener) Accept() (c net.Conn, err error) {
 	if err != nil {
 		return
 	}
-	kac := c.(keepAliveConn)
-	// detection time: tcp_keepalive_time + tcp_keepalive_probes + tcp_keepalive_intvl
-	// default on linux:  30 + 8 * 30
-	// default on osx:    30 + 8 * 75
-	kac.SetKeepAlive(true)
-	kac.SetKeepAlivePeriod(30 * time.Second)
```
Comment (ahrtr) [inline pkg/transport/keepalive_listener.go:80]: Why remove these?

File: pkg/transport/keepalive_listener.go [hunk 2]
```diff
@@ -72,17 +101,12 @@ func (l *tlsKeepaliveListener) Accept() (c net.Conn, err error) {
...
+
 	c = tls.Server(c, l.config)
 	return c, nil
 }
 
-// NewListener creates a Listener which accepts connections from an inner
+// newTLSKeepaliveListener creates a Listener which accepts connections from an inner
```
Comment (ahrtr) [inline pkg/transport/keepalive_listener.go:109]: minor comment: Let's try not make any unrelated change in backport PR, to avoid increase the review burden.

**Comment by ahrtr** [comment]
The original thought is to support OpenBSD with minimal and safe change. But It seems that we need to make non-trivial change in 3.4 to backport the PR. Since we do not officially support OpenBSD, so suggest to close this PR. Let's only backport the PR to 3.5.

Sorry for the confusion.

File: pkg/transport/keepalive_listener.go [hunk 3]
```diff
@@ -72,7 +101,7 @@ func (l *tlsKeepaliveListener) Accept() (c net.Conn, err error) {
 	if err != nil {
 		return
 	}
-	kac := c.(keepAliveConn)
+	kac := c.(*keepAliveConn)
```
Comment (Elbehery) [inline pkg/transport/keepalive_listener.go:104]: @ahrtr updated .. had to change `c.(keepAliveConn)` to `c.(*keepAliveConn)` cause the test case raised a panic till i fixed this 👍🏽

**Comment by Elbehery** [comment]
closing this since the code of 3.4 will have a breaking change
